### PR TITLE
Files OAuth

### PR DESCRIPTION
**Pull request by ibrahimrabab**
### Description

Adding **OAuth** support for File Shares as similarly done in Blobs and Queues.
Adding new enum: **ShareFileRequestIntent** which allows users to specify the intent for whether the file should be backed up.

### CODE REVIEW

File: sdk/storage/azure-storage-file-share/src/main/java/com/azure/storage/file/share/ShareClientBuilder.java [hunk 1]
```diff
@@ -10,12 +10,13 @@
 import com.azure.core.client.traits.ConnectionStringTrait;
 import com.azure.core.client.traits.EndpointTrait;
 import com.azure.core.client.traits.HttpTrait;
+import com.azure.core.client.traits.TokenCredentialTrait;
```
Comment (seanmcc-msft) [inline sdk/storage/azure-storage-file-share/src/main/java/com/azure/storage/file/share/ShareClientBuilder.java:13]: What about the DirectoryClientBuilder and LeaseClientBuilder?

File: sdk/storage/azure-storage-file-share/src/main/java/com/azure/storage/file/share/implementation/util/BuilderHelper.java [hunk 2]
```diff
@@ -3,12 +3,16 @@
 
 package com.azure.storage.file.share.implementation.util;
 
+import com.azure.core.credential.AzureSasCredential;
```
Comment (seanmcc-msft) [inline sdk/storage/azure-storage-file-share/src/main/java/com/azure/storage/file/share/implementation/util/BuilderHelper.java:6]: I may not understand the structure of the Java codebase very well, but is this a hand-written class?  If so, why is it an implementation folder?

Comment (ibrahimrabab) [reply sdk/storage/azure-storage-file-share/src/main/java/com/azure/storage/file/share/implementation/util/BuilderHelper.java:6]: Classes all under the implementation/util folder are all hand-written.

File: sdk/storage/azure-storage-file-share/src/main/java/com/azure/storage/file/share/ShareClientBuilder.java (hunk 1 above)
Comment (ibrahimrabab) [reply sdk/storage/azure-storage-file-share/src/main/java/com/azure/storage/file/share/ShareClientBuilder.java:13]: DirectoryClientBuilder does not exist, directoryClient can be created from the ShareFileClientBuilder (which I've added the new property to). And for LeaseClientBuilder, I've already added the new properties to ShareLeaseClientBuilder (under the specialized folder).

**Comment by seanmcc-msft** [comment]
Looking good.  Before merging, let's address the Arch Board's API view comments for Files OAuth.  This swagger PR contains the change renaming ShareFileRequestIntent -> ShareTokenIntent and making it an extensible enum - https://github.com/Azure/azure-rest-api-specs/pull/22699

**Review by seanmcc-msft — APPROVED** [review]

**Pull request merged**